_targets/build.project.riscv64-spike

```diff
@@ -54,37 +56,51 @@ b_build_target() {
 	# Relink kernel to include newly created cpio archive
 	(cd phoenix-rtos-kernel && make $MAKEFLAGS all)
 
+	# Convert kernel to flat binary
+	${CROSS}objcopy -O binary ${PREFIX_PROG}/phoenix-${TARGET}.elf ${PREFIX_PROG}/phoenix-${TARGET}.bin
+
 	# Add kernel to boot loader
 	b_log "Adding kernel to boot loader"
 
-	mkdir -p riscv-pk/build-${TARGET}
-	pushd riscv-pk/build-${TARGET}
-		OLDCFLAGS="$CFLAGS"
-		CFLAGS=""
-
-		if [ "X${CLEAN}" == "Xclean" ]; then
-
-			#make distclean
-			../configure --host=riscv64-phoenix-elf --with-payload=../../_build/riscv64-spike/prog/phoenix-riscv64-spike.elf --disable-fp-emulation
-		fi
-
-		make ${MAKEFLAGS} all
-		CFLAGS="$OLDCFLAGS"
-	popd
-
-	cp riscv-pk/build-${TARGET}/bbl $PREFIX_BOOT/phoenix-riscv64-spike.bbl
+	if [ "${RISCV_LOADER}" == "riscv-pk" ]; then
+		mkdir -p riscv/riscv-pk/build-${TARGET}
+		pushd riscv/riscv-pk/build-${TARGET}
+			if [ "X${CLEAN}" == "Xclean" ]; then
+				#make distclean
+				../configure --host=${CROSS%?} --with-payload=${PREFIX_PROG}/phoenix-${TARGET}.elf --disable-fp-emulation
+			fi
+
+			env -i PATH="$PATH" make ${MAKEFLAGS} all
+		popd
+	elif [ "${RISCV_LOADER}" == "opensbi" ]; then
+		pushd riscv/opensbi/
```

Comment:
build opensbi out-of-tree in `_build/$TARGET`- make proper use of `make O=<build_directory>`

Reply:
Fixed.